src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -1731,26 +1884,61 @@ subroutine wrfv3_netcdf(fv3filenamegin)
 
       call convert_cvpqx_to_qx(io_arr_qr, io_arr_qs, io_arr_qg, l_use_cvpqx, cvpqx_pval)  ! Convert Qx back
       call convert_cvpnx_to_nx(io_arr_qnr, l_cvpnr, cvpnr_pval, cld_nt_updt, ges_q, io_arr_qr, ges_ps) ! Convert Nx back 
+      ges_qr=io_arr_qr
+      ges_qs=io_arr_qs
+      ges_qg=io_arr_qg
+      ges_qnr=io_arr_qnr
+    endif
 
+    call gsi_copy_bundle(GSI_MetGuess_Bundle(it),gsibundle_fv3lam_dynvar_nouv) 
+    call gsi_copy_bundle(GSI_MetGuess_Bundle(it),gsibundle_fv3lam_tracer_nouv) 
+    call gsi_bundleputvar (gsibundle_fv3lam_dynvar_nouv,'tsen',ges_tsen(:,:,:,it),istatus)
+    if( fv3sar_bg_opt == 0) then
+      call GSI_BundleGetPointer ( gsibundle_fv3lam_dynvar_nouv, 'delp'  ,ges_delp ,istatus );ier=ier+istatus
+      allocate(g_prsi(lat2,lon2,nsig+1))
+      allocate(ges_ps_inc(lat2,lon2))
+      ges_ps_inc=ges_ps-ges_ps_bg 
+      g_prsi(:,:,nsig+1)=eta1_ll(nsig+1)
+      do i=nsig,1,-1
+      g_prsi(:,:,i)=ges_delp_bg(:,:,i)+g_prsi(:,:,i+1) 
+      enddo
+      do i=1,nsig+1
+        g_prsi(:,:,i)=g_prsi(:,:,i)+eta2_ll(i)*ges_ps_inc
+      enddo
+      do i=1,nsig
+        ges_delp(:,:,i)=g_prsi(:,:,i)-g_prsi(:,:,i+1)
+      enddo
+      ges_delp=ges_delp*1000.0_r_kind
+      deallocate(g_prsi,ges_ps_inc)
+    
+    else
+      ges_ps=ges_ps*1000.0_r_kind
+      call gsi_bundleputvar (gsibundle_fv3lam_dynvar_nouv,'ps',ges_ps,istatus)
+    endif
 !   write out
-      if( fv3sar_bg_opt == 0) then
-         add_saved=.true.
-         call gsi_fv3ncdf_write(dynvars,'W',ges_w,mype_w,add_saved)
-         call gsi_fv3ncdf_write(tracers,'liq_wat',ges_ql,mype_ql,add_saved)
-         call gsi_fv3ncdf_write(tracers,'ice_wat',ges_qi,mype_qi,add_saved)
-         call gsi_fv3ncdf_write(tracers,'rainwat',io_arr_qr,mype_qr,add_saved)
-         call gsi_fv3ncdf_write(tracers,'snowwat',io_arr_qs,mype_qs,add_saved)
-         call gsi_fv3ncdf_write(tracers,'graupel',io_arr_qg,mype_qg,add_saved)
-         call gsi_fv3ncdf_write(tracers,'rain_nc',io_arr_qnr,mype_qnr,add_saved)
-      else
-         write(6,*) "FV3 IO Write for 'fv3sar_bg_opt == 0 is only available for now in direct relfectivity DA"
-         stop
-      end if
-    end if
+    if (ier/=0) call die('get ges','cannot get pointers for fv3 met-fields, ier =',ier)
+
+    add_saved=.true.
+
+!   write out
+    if(fv3sar_bg_opt == 0) then
+      call gsi_fv3ncdf_write(grd_fv3lam_dynvar_ionouv,gsibundle_fv3lam_dynvar_nouv,add_saved,fv3filenamegin%dynvars,fv3filenamegin)
+      call gsi_fv3ncdf_write(grd_fv3lam_tracer_ionouv,gsibundle_fv3lam_tracer_nouv,add_saved,fv3filenamegin%tracers,fv3filenamegin)
+      call gsi_fv3ncdf_writeuv(grd_fv3lam_uv,ges_u,ges_v,add_saved,fv3filenamegin)
+
+
+    else
+      call gsi_fv3ncdf_write_v1(grd_fv3lam_dynvar_ionouv,gsibundle_fv3lam_dynvar_nouv,add_saved,fv3filenamegin%dynvars,fv3filenamegin)
+      call gsi_fv3ncdf_write_v1(grd_fv3lam_tracer_ionouv,gsibundle_fv3lam_tracer_nouv,add_saved,fv3filenamegin%tracers,fv3filenamegin)
```

Comment:
These lines exceed the 132 character limit. Please reduce the number of characters.